### ISSUE TITLE
Use samplings for the knots in b-splines

### ISSUE THREAD

**Issue by EmilyBourne** [CExA-project/ddc#416]
Now that #121 has been merged it is possible to have multiple samplings on the same dimension. This should be used to define the knots of splines. This will avoid the use of `uid()`

**Comment by tpadioleau** [comment]
I think we already use it ? Do you mean to use it outside the the `[Non]UniformBSplines` classes ? For now it is not exposed in the public API but could be.

**Comment by EmilyBourne** [comment]
Having looked at the code in more detail I think you are right. The knots are saved into a sampling, however the sampling I think it is possible to write the code in a way that uses this more explicitly to give more DDC-like code. I will propose a PR once #415 is merged

**Comment by tpadioleau** [comment]
Superseded by #432 and #459.
